src/session.ts

```diff
@@ -458,35 +458,57 @@ export class SessionManager implements Middleware {
         try {
             this.sessionDetails = await languageServerProcess.start("EditorServices");
         } catch (err) {
-            this.setSessionFailure("PowerShell process failed to start: ", err instanceof Error ? err.message : "unknown");
+            // We should kill the process in case it's stuck.
+            void languageServerProcess.dispose();
+
+            // PowerShell never started, probably a bad version!
+            const version = await languageServerProcess.getVersionCli();
+            let shouldUpdate = true;
+            if (satisfies(version, "<5.1.0")) {
+                void this.setSessionFailedGetPowerShell(`PowerShell ${version} is not supported, please update!`);
+            } else if (satisfies(version, ">=5.1.0 <6.0.0")) {
+                void this.setSessionFailedGetPowerShell("It looks like you're trying to use Windows PowerShell, which is supported on a best-effort basis. Can you try PowerShell 7?");
```

Comment:
Worth adding an aka.ms link to the install instructions here maybe?

Reply:
So with `this.setSessionFailedGetPowerShell` this actually now shows an error window popup with a button "Open PowerShell Install Documentation" that opens the link https://aka.ms/get-powershell-vscode!